vmm/src/vm_config.rs

```diff
@@ -428,6 +428,11 @@ pub struct ConsoleConfig {
     #[serde(default)]
     pub iommu: bool,
     pub socket: Option<PathBuf>,
+    // main_fd is the fd of the pre-created console device
```

Comment:
I'm afraid it's not correct to put state like this in a `Config` struct - it's for user supplied configuration only.